# Introduction

Microbiome analysis is here to stay, but there are challenges to interpreting this type of data that may not be apparent at first glance.  In this section of the course we are going to cover everything you need to start peforming microbiome analyses, including a look at some insidious pitfalls along the way.

# Learning Objectives

### Microbiome Sequence Clustering

By the end of class today you should be able to comfortably answer these questions:
1.	Explain what a 'molecular diagnostic' is in the context of microbiome analysis and how does that differ from a traditional microbial presence assay?
2.	Define what gene is most commonly used to identify if an organism is present in some sample?
3.	Name three important characteristics that explain why this gene is used.
4.	Understand the basic laboratory process required to sequence a microbiome.
5.	Define PCR, and be able to describe the steps.
6.	Identify some potential pitfalls one could encounter because of the PCR step?
7.	Discuss how properties of the 16S gene can be problematic in DNA sequencing.
8.	Define an 'OTU' and why have people traditionally used it, rather than simply counting identical sequences?
9.	Classify the important filetypes to know and understand for OTU clustering?
10.	Understand how OTU clustering is performed, and execute on sequence data.
11.	Understand and perform taxonomic assignment.
12.	Define the steps required to create an OTU table.
13. Differentiate alpha, beta, and gamma diversity metrics and discuss what they attempt to measure? (maybe next week)

# Reading Material

Over the next few classes we will be using the [usearch tool](https://www.drive5.com) fairly extensively

# Assignment

Todays assignment is as much about learning to use a popular microbiome toolset, as it is a continued tutorial in how to use linux effectively.  Get excited because we will be using some tools you may not have encountered, and some of them might change how you think about/interact with the linux operating system!